src/lib.rs

```diff
@@ -352,6 +352,86 @@ impl Filter {
     }
 }
 
+/// The possible input methods.
+pub enum InputMethod {
+    /// Read the input from the local Git history.
+    GitHistory,
+
+    /// Read the specified input file.
+    LogFile(String),
+
+    /// There was no input method specified.
+    None,
```

Comment:
I would remove this and let `new` return an option. This moves error handling to the beginning of execution.

Reply:
Due to our previous discussion on failing constructors I would like to rename it additionally to `parse` in order to keep the naming uniform. :wrench:

Reply:
:wrench: :heavy_check_mark: